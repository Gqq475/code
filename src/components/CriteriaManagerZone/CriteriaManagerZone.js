import React from 'react'
import classes from './CriteriaManagerZone.scss'
import { Switch, Slider, Button } from '@blueprintjs/core'
import Left from './DetailLeft'
import Center from './DetailCenter'
import Right from './DetailRight'

type Props = {
  data: Object
};

export class CriteriaManagerZone extends React.Component {
  props: Props

  constructor (props) {
    super(props)
    this.handleChange = this.handleChange.bind(this)
    this.handleSliderChange = this.handleSliderChange.bind(this)
    this.handleEnabledChange = this.handleEnabledChange.bind(this)
    this.state = {
      alarmLevel: 3
    }
  }

  handleChange (type, e) {
    // const { pageDetail } = this.state
    // console.log('pageDetail', e.target.value)
    // pageDetail[type] = e.target.value
    // console.log(pageDetail)
    // this.setState = {
    //   pageDetail: pageDetail
    // }
  }

  handleEnabledChange () {
    // const { pageDetail } = this.state
    // pageDetail['enabled'] = !pageDetail['enabled']
    // this.setState({
    //   pageDetail: pageDetail
    // })
  }

  handleSliderChange (value) {
    this.setState({ alarmLevel: value })
  }

  render () {
    let { data } = this.props
    return (
      <div className={classes['CriteriaManagerZone-container']}>
        <span className={classes['header']}>
          <div className={classes.info}>
            <span>规则详情</span>
            <strong>ID {data.id}</strong>
          </div>
          <div>
            <Button className='pt-intent-primary'
              onClick={this.handleEditChange}>编辑</Button>
          </div>
        </span>
        <section className={classes.top}>
          <div className={classes['enabled-container']}>
            <Switch className='pt-large'
              label='启用'
              defaultChecked={data ? data.enabled : false}
              onChange={this.handleEnabledChange} />
          </div>
          <div className={classes['slider-container']}>
            <span>{_.ALARMLEVEL} </span>
            <Slider
              min={1}
              max={9}
              labelStepSize={1}
              onChange={this.handleSliderChange}
              value={this.state.alarmLevel} />
          </div>
          <div className={classes['alarmAmount']}>
            <strong>10</strong> 告警
          </div>
        </section>
        <div className={classes.content}>
          <Left data={data} />
          <Center data={data} />
          <Right data={data} />
        </div>
      </div>
    )
  }
}

export default CriteriaManagerZone
