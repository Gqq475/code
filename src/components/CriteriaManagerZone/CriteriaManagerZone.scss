$border: solid 1px #d7d7d7;

.CriteriaManagerZone-container {
  border: solid 1px #0073bb;
  margin: 20px 0;
  border-radius: 5px;
  overflow: hidden;

  .header {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 10px 20px 5px;
    border-bottom: solid 1px #d7d7d7;
    background: #f7f7f7;
  }

  .info {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: flex-start;
    letter-spacing: .5px;
  }
}

.top {
  display: flex;
  justify-content: space-between;
  align-items: center;

  .enabled-container,
  .alarmAmount {
    margin: 0 auto;
  }

  .enabled-container {
    height: 40px;
    display: flex;
    align-items: flex-end;
  }

  .slider-container {
    width: 50%;
    padding: 0 50px;
    height: 80px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    border-left: $border;
    border-right: $border;
  }
}

.content {
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  background: #f7f7f7;
  border-top: $border;

  .title {
    font-size: 15px;
    margin-bottom: 20px;
    font-weight: 600;
  }

  .title::first-letter {
    font-size: 18px;
  }
}

$content-padding: 20px 20px 10px 20px;

.left-container {
  width: 33.33%;
  padding: $content-padding;
  padding-right: 40px;
}

.center-container {
  border-left: $border;
  border-right: $border;
  width: 33.33%;
  padding: $content-padding;

  .ul {
    padding-left: 20px;
  }
  .ul::before {
    display: none;
  }

  .total-defect {
    margin: 10px 0;
  }

  .checkbox-group {
    margin: 0 10px 5px 0;
  }
}

.right-container {
  width: 33.33%;
  padding: $content-padding;

  .time-group-container {
    margin: 5px 0 0 20px;
    width: 260px;
    display: flex;
    justify-content: flex-start;
    align-items: center;
  }

  .time-input {
    width: 80px;
  }

  .time-input-text {
    margin: 0 10px;
  }
}
