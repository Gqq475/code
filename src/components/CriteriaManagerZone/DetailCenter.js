import React from 'react'
import classes from './CriteriaManagerZone.scss'
import { Tabs, Tab, TabList, TabPanel, Checkbox } from '@blueprintjs/core'

type Props = {
  data: Object
};

export class DetailCenter extends React.Component {
  props: Props

  constructor (props) {
    super(props)
    this.handleChange = this.handleChange.bind(this)
    this.handleDefectClick = this.handleDefectClick.bind(this)
    this.state = {
      qualityJson: {
        totalDefect: true,
        defectSize: false
      }
    }
  }

  handleDefectClick () {
  }

  handleChange () {
  }

  render () {
    let { data } = this.props

    const defectSize = ['SS', 'S', 'M', 'L', 'O']

    return (
      <div className={classes['center-container']}>
        <section className={classes.title}>2. Condition</section>
        <label className='pt-label pt-inline'>
          OOC
          <input className='pt-input'
            defaultValue={data.ooc}
            onChange={this.handleChange('OOC')} />
        </label>
        <label className='pt-label pt-inline'>
          OOS
          <input className='pt-input'
            defaultValue={data.oos}
            onChange={this.handleChange('OOS')} />
        </label>
        <Tabs>
          <TabList>
            <Tab>Defect</Tab>
            <Tab isDisabled={!false}>Burr</Tab>
            <Tab isDisabled={!false}>NG/RW Rate</Tab>
            <Tab isDisabled={!false}>LOI</Tab>
            <Tab isDisabled={!false}>CCD</Tab>
          </TabList>
          <TabPanel>
            <div className='pt-select'>
              <select>
                <option value='1'>扫边</option>
                <option value='2'>Common defect</option>
                <option value='3'>Cluster defect</option>
                <option value='4'>Mask defect</option>
                <option value='5'>定制化条件</option>
              </select>
            </div>
            <div className={classes['total-defect']}>
              <Checkbox
                defaultChecked={this.state.qualityJson.totalDefect}
                onChange={this.handleDefectClick('totalDefect')}>
                Total defect
              </Checkbox>
              <ul className={classes.ul}>
                <li className={classes['defect-size']}>
                  <Checkbox
                    defaultChecked={this.state.qualityJson.defectSize}
                    disabled={!this.state.qualityJson.totalDefect}
                    onChange={this.handleDefectClick('defectSize')}>
                    Defect size
                  </Checkbox>
                  <ul className={classes['ul']}>
                    {
                      defectSize.map((s, i) =>
                        <Checkbox
                          key={i}
                          className={`pt-inline ${classes['checkbox-group']}`}
                          disabled={!this.state.qualityJson.totalDefect}
                          defaultChecked={this.state.qualityJson[s]}
                          onChange={this.handleDefectClick(s)}>
                          {s}
                        </Checkbox>)
                    }
                  </ul>
                </li>
                <li>
                  <label className='pt-label'>
                    <Checkbox
                      disabled={!this.state.qualityJson.totalDefect}
                      checked={this.state.qualityJson.defectCode}
                      onChange={this.handleDefectClick('defectCode')}>
                      Defect Code (用逗号分隔)
                    </Checkbox>
                    <input
                      className='pt-input'
                      disabled={!this.state.qualityJson.totalDefect}
                      defaultValue={this.state.qualityJson.defectCodeValue}
                      onChange={this.handleDefectClick('defectCodeValue')} />
                  </label>
                </li>
                <li>
                  <label className='pt-label'>
                    <Checkbox
                      disabled={!this.state.qualityJson.totalDefect}
                      checked={this.state.qualityJson.judegDefectCode}
                      onChange={this.handleDefectClick('judegDefectCode')}>
                      Judge defect Code (用逗号分隔)
                    </Checkbox>
                    <input
                      className='pt-input'
                      disabled={!this.state.qualityJson.totalDefect ||
                        !this.state.qualityJson.judegDefectCode}
                      defaultValue={this.state.qualityJson.judegDefectCodeValue}
                      onChange={this.handleDefectClick('judegDefectCodeValue')} />
                  </label>
                </li>
              </ul>
            </div>
          </TabPanel>
          <TabPanel />
          <TabPanel />
          <TabPanel />
          <TabPanel />
        </Tabs>
      </div>
    )
  }
}

export default DetailCenter
