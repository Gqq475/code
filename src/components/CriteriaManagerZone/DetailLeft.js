import React from 'react'
import classes from './CriteriaManagerZone.scss'
// import {  } from '@blueprintjs/core'

type Props = {
  data: Object
}

export class DetailLeft extends React.Component {
  props: Props

  constructor (props) {
    super(props)
    this.handleChange = this.handleChange.bind(this)
  }

  handleChange () {

  }

  render () {
    let { data } = this.props
    return (
      <div className={classes['left-container']}>
        <section className={classes.title}>1. Product</section>
        <section>
          <label className='pt-label'>
            Product
            <div className='pt-select'>
              <select
                onChange={this.handleChange('productName')}>
                <option value='SV509-CF'>SV509-CF</option>
                <option value='JV508-CG'>JV508-CG</option>
                <option value='GV507-DF'>GV507-DF</option>
                <option value='NV506-HF'>NV506-HF</option>
                <option value='8V505-BF'>8V505-BF</option>
              </select>
            </div>
          </label>
        </section>
        <section>
          <label className='pt-label'>
            Operation
            <div className='pt-select'>
              <select
                onChange={this.handleChange('operationName')}>
                <option value='2100'>2100</option>
                <option value='3100'>3100</option>
                <option value='4100'>4100</option>
                <option value='5100'>5100</option>
                <option value='6100'>6100</option>
              </select>
            </div>
          </label>
        </section>
        <section>
          <label className='pt-label'>
            LINE ID
            <div className='pt-select'>
              <select
                onChange={this.handleChange('lineIDName')}>
                <option value='PIPR01'>PIPR01</option>
                <option value='KIHR02'>KIHR02</option>
                <option value='YIHR10'>YIHR10</option>
                <option value='LIPK90'>LIPK90</option>
                <option value='OIPK01'>OIPK01</option>
              </select>
            </div>
          </label>
        </section>
        <section>
          <label className='pt-label'>
            Sub entity
            <div className='pt-select'>
              <select
                onChange={this.handleChange('subEntityName')}>
                <option value='PIN01'>PIN01</option>
                <option value='PIN02'>PIN02</option>
                <option value='PIN03'>PIN03</option>
                <option value='PIN04'>PIN04</option>
                <option value='PIN05'>PIN05</option>
              </select>
            </div>
          </label>
        </section>
        <section>
          <label className='pt-label'>
            Outliner
            <input className='pt-input'
              type='text'
              placeholder=''
              dir='auto'
              defaultValue={data.outliner}
              onChange={this.handleChange('outliner')} />
          </label>
        </section>
      </div>
    )
  }
}

export default DetailLeft
