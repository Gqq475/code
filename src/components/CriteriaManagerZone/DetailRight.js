import React from 'react'
import classes from './CriteriaManagerZone.scss'
import { Radio } from '@blueprintjs/core'

type Props = {
  data: Object
}

export class DetailRight extends React.Component {
  props: Props

  constructor (props) {
    super(props)
    this.handleTimeConditionRChange = this.handleTimeConditionRChange.bind(this)
    this.state = {
      timeCondPieceType: 1
    }
  }

  handleTimeConditionRChange () {
  }

  render () {
    return (
      <div className={classes['right-container']}>
        <div className={classes.title}>
          3. Time condition
        </div>
        <Radio label='单片超过（无时间限定）'
          defaultChecked={this.state.timeCondPieceType === '1'}
          name='condition'
          value='1'
          onChange={this.handleTimeConditionRChange('1')} />
        <Radio label='时间区间内设定'
          name='condition'
          value='2'
          defaultChecked={this.state.timeCondPieceType === '2'}
          onChange={this.handleTimeConditionRChange('2')} />
        <div className={classes['time-group-container']}>
          <div className='pt-control-group'>
            <div className='pt-select'>
              <select>
                <option value='1'>固定区间总和</option>
                <option value='2'>固定区间平均</option>
                <option value='3'>移动区间总和</option>
                <option value='4'>移动区间平均</option>
              </select>
            </div>
            <div className={`pt-input-group ${classes['time-input']}`}>
              <input className='pt-input' />
            </div>
          </div>
          <span className={classes['time-input-text']}>分钟</span>
        </div>
        <div className={classes['time-group-container']}>
          <span className={classes['time-input-text']}>最少</span>
          <div className='pt-control-group'>
            <div className='pt-select'>
              <select>
                <option value='1'>连续片数</option>
                <option value='2'>累积片数</option>
                <option value='3'>平均片数</option>
              </select>
            </div>
            <div className={`pt-input-group ${classes['time-input']}`}>
              <input className='pt-input' />
            </div>
          </div>
          <span className={classes['time-input-text']}>片</span>
        </div>
      </div>
    )
  }
}

export default DetailRight
