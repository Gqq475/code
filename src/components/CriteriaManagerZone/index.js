import CriteriaManagerZone from './CriteriaManagerZone'
export default CriteriaManagerZone
