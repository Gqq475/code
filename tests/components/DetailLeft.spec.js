import React from 'react'
import { DetailLeft } from 'components/CriteriaManagerZone/DetailLeft'

describe('(Component) DetailLeft', () => {
  it('render normally', () => {
    const data = {
      'outliner': '1'
    }
    const wrapper = shallow(
      <DetailLeft data={data} />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
