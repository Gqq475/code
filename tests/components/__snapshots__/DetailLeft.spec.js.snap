// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) DetailLeft render normally 1`] = `
<div
  className="left-container"
>
  <section
    className="title"
  >
    1. Product
  </section>
  <section>
    <label
      className="pt-label"
    >
      Product
      <div
        className="pt-select"
      >
        <select>
          <option
            value="SV509-CF"
          >
            SV509-CF
          </option>
          <option
            value="JV508-CG"
          >
            JV508-CG
          </option>
          <option
            value="GV507-DF"
          >
            GV507-DF
          </option>
          <option
            value="NV506-HF"
          >
            NV506-HF
          </option>
          <option
            value="8V505-BF"
          >
            8V505-BF
          </option>
        </select>
      </div>
    </label>
  </section>
  <section>
    <label
      className="pt-label"
    >
      Operation
      <div
        className="pt-select"
      >
        <select>
          <option
            value="2100"
          >
            2100
          </option>
          <option
            value="3100"
          >
            3100
          </option>
          <option
            value="4100"
          >
            4100
          </option>
          <option
            value="5100"
          >
            5100
          </option>
          <option
            value="6100"
          >
            6100
          </option>
        </select>
      </div>
    </label>
  </section>
  <section>
    <label
      className="pt-label"
    >
      LINE ID
      <div
        className="pt-select"
      >
        <select>
          <option
            value="PIPR01"
          >
            PIPR01
          </option>
          <option
            value="KIHR02"
          >
            KIHR02
          </option>
          <option
            value="YIHR10"
          >
            YIHR10
          </option>
          <option
            value="LIPK90"
          >
            LIPK90
          </option>
          <option
            value="OIPK01"
          >
            OIPK01
          </option>
        </select>
      </div>
    </label>
  </section>
  <section>
    <label
      className="pt-label"
    >
      Sub entity
      <div
        className="pt-select"
      >
        <select>
          <option
            value="PIN01"
          >
            PIN01
          </option>
          <option
            value="PIN02"
          >
            PIN02
          </option>
          <option
            value="PIN03"
          >
            PIN03
          </option>
          <option
            value="PIN04"
          >
            PIN04
          </option>
          <option
            value="PIN05"
          >
            PIN05
          </option>
        </select>
      </div>
    </label>
  </section>
  <section>
    <label
      className="pt-label"
    >
      Outliner
      <input
        className="pt-input"
        defaultValue="1"
        dir="auto"
        placeholder=""
        type="text"
      />
    </label>
  </section>
</div>
`;
